rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -64,6 +65,11 @@ struct FromGenericMetricsMessage {
     aggregation_option: Option<String>,
 }
 
+#[derive(Debug, Deserialize)]
+struct MessageUseCase {
+    use_case_id: String,
+}
```

Comment:
@Dav1dde I am not familiar with the serde create to know if this is "fast enough". I think you're the best person to comment on the approach here.

The killswitch will be inactive most of the time, so deserializing this struct needs to be fast enough to justify this additional step during normal operation. Otherwise I would prefer it if we just check the `use_case_id` field of the fully deserialized struct.

Reply:
It will definitely go through the entire data an additional time, it does not stop early, that being said it's also not fully deserializing the entire JSON.

What should work is:

```
    let msg: FromGenericMetricsMessage = if let Some(killswitch) = killswitch_config.ok().flatten() {
        let data: serde_json::Value = serde_json::from_slice(payload_bytes)?;
        let use_case_id = data.as_object().and_then(|obj| obj.get("use_case_id"));
        if Some(killswitch) == use_case.use_case_id {
            return Ok(InsertBatch::skip());
        }
        let msg = serde_json::from_value(data)?;
    } else {
        serde_json::from_slice(payload_bytes)?;
    }
```

But this is getting quite complicated, does having the killswitch provide us any other benefit over just having a DLQ?